.github/workflows/cd.yaml

```diff
@@ -10,142 +10,199 @@ on:
-          sudo apt-get install -y --no-install-recommends libsdl2-dev libpng-dev libjpeg-turbo8-dev libopenal-dev libmad0-dev libglew-dev libgl1-mesa-dev libegl1-mesa-dev libgles2-mesa-dev uuid-dev
-          python3 -m pip install --user scons
-      - name: Adjust version strings
-        run: ./utils/cd_update_versions.sh
-        shell: bash
-      - name: Build Application
-        run: scons -Qj $(nproc)
-      - name: Run Unit Tests
-        run: scons -Qj $(nproc) test
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Cache ccache results
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ubuntu-20.04-desktop-ccache-${{ github.repository }}-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            ubuntu-20.04-desktop-ccache-${{ github.repository }}-${{ github.ref }}-
+            ubuntu-20.04-desktop-ccache-${{ github.repository }}-
+            ubuntu-20.04-desktop-ccache-
+      - name: Get Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install development dependencies
+        uses: ./.github/dependencies
+      - uses: lukka/get-cmake@latest
+      - uses: lukka/run-vcpkg@v10
+        with:
+          vcpkgJsonGlob: 'vcpkg.json'
+      - uses: lukka/run-cmake@v10
+        with:
+          configurePreset: 'linux-ci'
+          buildPreset: 'linux-ci'
+          testPreset: 'linux-cd'
       - name: Package Application
-        run: tar -czf ${{ env.OUTPUT }} sounds images/ data/ license.txt keys.txt icon.png endless-sky credits.txt copyright changelog
+        run: ./utils/build_appimage.sh build/ci
       - name: Upload artifact
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
           name: ${{ env.OUTPUT }}
           path: ${{ env.OUTPUT }}
 
-  cd_appimage_x86_64:
-    runs-on: ubuntu-18.04
+
+  windows_win32:
+    name: Windows x86
+    runs-on: windows-latest
     env:
-      ARCH: x86_64
-      OUTPUT: endless-sky-x86_64-continuous.AppImage
+      OUTPUT: EndlessSky-win32-continuous.zip
+      GCC: 12.1.0
+      CCACHE_DIR: ${{ github.workspace }}/ccache
     steps:
-      - uses: actions/checkout@v2
-      - name: Install dependencies
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Cache ccache results
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ env.GCC }}-x86-ccache-${{ github.repository }}-${{ github.ref }}-${{ github.sha }}
+          restore-keys: |
+            ${{ env.GCC }}-x86-ccache-${{ github.repository }}-${{ github.ref }}-
+            ${{ env.GCC }}-x86-ccache-${{ github.repository }}-
+            ${{ env.GCC }}-x86-ccache
+      - name: Install development dependencies
+        uses: ./.github/dependencies
+      - name: Get MinGW
+        uses: quyykk/get-winlibs-mingw@v1
```

Comment:
A good practice in software engineering is not to run production code form your own personal home-directory (or someone elses home-directory). This is technically a little bit different, since it is a published repository, but I still prefer to host this action under endless-sky, not under your own personal github site, if we need it in the build-process.

Reply:
> I think I already asked about the lukka actions below, but also there; could we clone/fork them into endless-sky and run them from our own clone/fork?

I don't really see why this is necessary tbh. We wouldn't get any advantage from doing this, because we would just keeping our fork in sync with upstream anyways. Might as well just use the upstream directly.

> A good practice in software engineering is not to run production code form your own personal home-directory (or someone elses home-directory)

I don't think this analogy holds. Github Actions were made to be reusable, so that people can share their actions with others. I think a better analogy is it's like using an external library (SDL, libmad, libpng).

I wouldn't mind moving my action over to this org.

Reply:
Please do move over the action; it just looks more professional. (And it is a shame that Lukka is creating much better vcpkg installers than that Microsoft is doing on their official repos.)